### PR TITLE
Optimize and refactor currency parser for improved performance and clarity 🚀

### PR DESCRIPTION
**Pull request by mohamedmamdouh22**
- Enhanced the `ParserTypeCurrency` class by optimizing currency and unavailable price phrase lookup times using sets for O(1) complexity.
- Added the currency symbol at the beginning for more clarity

### CODE REVIEW

File: harambe/parser/type_currency.py [hunk 1]
```diff
@@ -3,68 +3,110 @@
...
+        "₼",
+        "₢",
+        "₣",
+        "₥",
+        "₯",
+        "₠",
+        "₧",
+        "₰",
+    ]
+)
```
Comment (awtkns) [inline harambe/parser/type_currency.py:65]: These shouldnt be in an array

```suggestion
price_not_available_phrases = {
    "price not available",
    "unavailable price",
    "price upon request",
    "contact for price",
    "request a quote",
    "call for price",
    "check price in store",
    "price tbd",
    "price not disclosed",
    "out of stock",
    "sold out",
    "pricing not provided",
    "not priced",
    "currently unavailable",
    "n/a",
    "ask for pricing",
    "see details for price",
    "price coming soon",
    "temporarily unavailable",
    "price hidden",
    "tdb"
}

currency_symbols = {
    "$", "€", "£", "¥", "₹", "₽", "₩", "₫", "₴", "₦", "₪", "₱", "฿", "₲", "₭", 
    "₵", "₡", "₸", "₺", "ƒ", "₳", "₼", "₢", "₣", "₥", "₯", "₠", "₧", "₰"
}
```

File: test/parser/test_type_currency.py [hunk 2]
```diff
@@ -11,31 +11,35 @@ class _TestModel(BaseModel):
...
+        ("€1.234", "€1234.0"),
+        ("1,234$", "$1234.0"),
+        ("1,234 $", "$1234.0"),
+        ("1.234,45", "1234.45"),
+        ("1.234.456,00", "1234456.0"),
+        ("1.000.000", "1000000.0"),
+        ("Starting At 12.99", "12.99"),
+        ("From 399.99", "399.99"),
+        ("From 399.99$", "$399.99"),
+        ("0.0004 $", "$0.0004"),
```
Comment (awtkns) [inline test/parser/test_type_currency.py:42]: Personally i think currecy should be outputed as an array:
{
   "currency": "USD"
   "price": 12.99
}

Comment (awtkns) [reply test/parser/test_type_currency.py:42]: Screenshot 2024-10-02 at 3 47 55 PM
 example from zyte

Comment (mohamedmamdouh22) [reply test/parser/test_type_currency.py:42]: hmmm yeah, asim and i discussed this and we will create a new ticket for this once agreed on the best object representation

**Pull request closed without merging**